dds/DCPS/RTPS/Sedp.h

```diff
@@ -760,27 +765,48 @@ class Sedp : public DCPS::EndpointManager<ParticipantData_t> {
 
     virtual ~TypeLookupReplyReader();
 
+    void get_continuation_point(const GuidPrefix_t& guid_prefix,
+                                const XTypes::TypeIdentifier& remote_ti,
+                                XTypes::OctetSeq32& cont_point) const;
+
+    void cleanup(const XTypes::TypeIdentifier& ti);
+
   private:
     virtual void data_received_i(const DCPS::ReceivedDataSample& sample,
       const DCPS::EntityId_t& entity_id,
       DCPS::Serializer& ser,
       DCPS::Extensibility extensibility);
 
-    DDS::ReturnCode_t process_tl_reply(DCPS::Serializer& ser);
-  };
+    DDS::ReturnCode_t process_type_lookup_reply(const DCPS::ReceivedDataSample&,
+                                                DCPS::Serializer& ser,
+                                                bool is_discovery_protected);
+    DDS::ReturnCode_t process_get_types_reply(const XTypes::TypeLookup_Reply&);
+    DDS::ReturnCode_t process_get_dependencies_reply(const DCPS::ReceivedDataSample&,
+                                                     const XTypes::TypeLookup_Reply&,
+                                                     const DCPS::SequenceNumber&,
+                                                     bool is_discovery_protected);
 
-  typedef DCPS::RcHandle<TypeLookupReplyReader> TypeLookupReplyReader_rch;
+    typedef std::pair<XTypes::OctetSeq32, XTypes::TypeIdentifierSeq> ContinuationPair;
 
-  TypeLookupRequestReader_rch type_lookup_request_reader_;
+    // Map from each remote type to the most recent continuation_point
+    // and all dependencies received for that type so far.
+    typedef OPENDDS_MAP(XTypes::TypeIdentifier, ContinuationPair) RemoteDependencies;
+
+    // Map from each remote participant to the data stored for its types.
+    typedef OPENDDS_MAP(GuidPrefixWrapper, RemoteDependencies) DependenciesMap;
+    DependenciesMap dependencies_;
```

Comment:
Do you mean the parent `Reader` class? If so, I think no since this map is specific to type lookup service (more specifically to getTypeDependencies of type lookup service).

Reply:
How about TypeLookupService class?

Reply:
Note (from standup): we'll keep it in this class for now.